keps/sig-node/4742-node-topology-downward-api/README.md

```diff
@@ -0,0 +1,417 @@
+
+Check fieldRefs referenced in Pod API.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason: 
+- [X] API .status
+  - Condition name: 
+  - Other field: 
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+The reliability of topology information for downward API should be the same as other fieldRefs
+available through downward API. There is no existing SLOs to measure the reliability of the downward API.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+
+- [ ] Metrics
+  - Metric name:
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [X] Other (treat as last resort)
+  - Details: Application-level observability
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A
+
+### Dependencies
+
+N/A
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Negligible increase to Pod size.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+Same as any existing field references used with downward API. 
+
+###### What are other known failure modes?
+
+Not all Kubernetes clusters have nodes that have topology labels. Requesting topology information
+via downward API in these clusters will result in Pods failing to start or empty values returned
+via downward API.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+Revert feature gate and stop consuming downward API. 
+
+## Implementation History
+
+- the `Summary`, `Motivation`, `Design Details` sections being merged, signaling SIG acceptance
+
+## Drawbacks
+
+Topology information is a Node-level construct. By implementing this KEP we are allowing information from Nodes
+to trickel down to Pods. Expanding this pattern to additional node information in the future can lead to
+weaker security boundaries between Pods and Nodes.
+
+## Alternatives
+
```

Comment:
As an alternative (and remember, alternatives are the choices we considered but _didn't_ pick): a controller that copies selected node labels to Pods (either as the same label exactly, or storing an annotation based on that subset of node labels).

This would make the information available but depending on exactly how we describe it, you might need a helper entrypoint to decode it. Also, somebody needs to write the controller and release it.